lib/generator.js

```diff
@@ -275,22 +273,65 @@ class Generator {
    * const Generator = require('asyncapi-generator');
    * const content = await Generator.getTemplateFile('html', '.partials/content.html');
    *
-   * @example <caption>Obtaining the content of a file from a custom template</caption>
-   * const path = require('path');
-   * const Generator = require('asyncapi-generator');
-   * const content = await Generator.getTemplateFile('myTemplate', 'a/file.js', {
-   *   templatesDir: path.resolve(__dirname, 'my-templates')
-   * });
-   *
    * @static
    * @param {String} templateName  Name of the template to generate.
    * @param {String} filePath      Path to the file to render. Relative to the template directory.
    * @param {Object} options
-   * @param {String} [options.templatesDir]  Path to the directory where to find the given template. Defaults to internal `templates` directory.
    * @return {Promise}
    */
-  static async getTemplateFile(templateName, filePath, { templatesDir } = {}) {
-    return await readFile(path.resolve(templatesDir || DEFAULT_TEMPLATES_DIR, templateName, filePath), 'utf8');
+  static async getTemplateFile(templateName, filePath) {
+    return await readFile(path.resolve(DEFAULT_TEMPLATES_DIR, templateName, filePath), 'utf8');
+  }
+
+  /**
+   * Downloads and installs a template and its dependencies.
+   *
+   * @param {Boolean} [force=false] Whether to force installation (and skip cache) or not.
+   */
+  installTemplate(force = false) {
+    return new Promise(async (resolve, reject) => {
+      if (!force) {
+        try {
+          let installedPkg;
+
+          if (isFileSystemPath(this.templateName)) {
+            const pkg = require(path.resolve(this.templateName, 'package.json'));
+            installedPkg = require(path.resolve(DEFAULT_TEMPLATES_DIR, pkg.name, 'package.json'));
+            if (installedPkg.name !== pkg.name) throw new Error('Package not installed');
+          } else { // Template is not a filesystem path...
+            const templatePath = path.resolve(DEFAULT_TEMPLATES_DIR, this.templateName);
+            if (await isLocalTemplate(templatePath)) {
+              const { resolvedLink } = await getLocalTemplateDetails(templatePath);
+              console.warn(`WARNING: this template is already installed and pointing to your filesystem at ${resolvedLink}.`);
```

Comment:
Not sure if this should be a warning or an "info" message.

Reply:
Also, it should probably say "Skipping installation..." or something similar. What do you think?

Reply:
What about we add a info message before searching locally and then another info message here that it was found locally and you should use the force install flag to update the template package 🤔 ?

Reply:
tbh I'm not sure exactly when will I get this warning anyway? How can I have a situation that remote template is actually local?

Reply:
For instance, imagine we're working locally on our HTML template and you run:

```
ag asyncapi.yaml ~/html-template
```

That template contains a package.json with the name `@asyncapi/html-template` and therefore npm will install it as `node_modules/@asyncapi/html-template`. However, because it's a local template, it will use a soft link pointing to `~/html-template`.

Right afterward, you run:

```
ag asyncapi.yaml @asyncapi/html-template
```

And the generator finds that the template has been already installed **but** it's a symlink and therefore it knows it's a local template, even though the name (`@asyncapi/html-template`) doesn't look like it is. This is that use case 😄

Reply:
well explained, thanks, makes a lot of sense. This is `info` for me, not a `warning` then